dffml/operation/archive.py

```diff
@@ -0,0 +1,74 @@
+import uuid
+from zipfile import ZipFile
+from pathlib import Path
+from typing import Union, Dict, Any, Optional
+
+from ..df.base import op
+from ..df.types import Definition
+
+
+# definitions
+PATH = Definition(name="path", primitive="Union[Path, str]")
+OPTIONS = Definition(name="options", primitive="Dict[str, Any]")
+
+
+@op(
+    inputs={
+        "input_directory_path": PATH,
+        "output_file_path": PATH,
+        "options": OPTIONS,
```

Comment:
Let's use different definitions for `input_directory_path` and `output_file_path`.

We want to make definitions for inputs as specific as possible. Think about it as being a piece of data that could be used in the same way by another operation. For example, your unzip operation would take a zip archive, however it wouldn't be able to unpack a directory, therefore we know that definition of a `PATH` is not specific.

```suggestion
        "input_directory_path": DIRECTORY,
        "output_file_path": ZIP_ARCHIVE,
```

Reply:
Updated